content/en/docs/ops/configuration/traffic-management/network-topologies/index.md

```diff
@@ -0,0 +1,190 @@
+
+To simplify configuring network topology during installation create a single YAML file to pass to `istioctl`:
+
+{{< text yaml >}}
+cat <<'EOF' > topology.yaml
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      gatewayTopology:
+EOF
+{{< /text >}}
+
+{{< idea >}}
+If Istio ingress gateway was already running prior to application of the `MeshConfig` you will need
+to restart any Istio ingress gateway pods.
+{{< /idea >}}
+
+Both of the settings discussed can also be configured using the `proxy.istio.io/config` annotation to the Pod spec
+of your Istio ingress gateway.
+
+{{< text yaml >}}
+...
+  metadata:
+    annotations:
+      "proxy.istio.io/config": '{"gatewayTopology" : { "numTrustedProxies": 2 } }'
+{{< /text >}}
+
+### Configuring X-Forwarded-For Headers
+
+Applications rely on client attributes, such as `X-Forward-For` header,
+to be forwarded by reverse proxies in a request. However, due to the variety of network
+topologies Istio can be deployed in, it is required to set the number of trusted proxies deployed in front
+of the Istio gateway proxy so that the client address can be extracted correctly.
+
+To set the number of trusted proxies add the following to your `topology.yaml` file.
+
+{{< text yaml >}}
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      gatewayTopology:
+        numTrustedProxies: <VALUE>
+{{< /text >}}
+
+For example, if you have a cloud based Load Balancer, a reverse proxy and then the Istio gateway proxy
+then `<VALUE>` would be 2.
+
+{{< idea >}}
+Note that all the proxies in front need of Istio gateway proxy must parse HTTP traffic and append the `X-Forwarded-For`
+headers at each hop. If the number of entries in `X-Forwarded-For` header is less than the number of
+trusted hops configured, Envoy falls back to using the immediate downstream address as the trusted
+client address. Please refer to [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for)
+to understand how `X-Forwarded-For` headers and trusted client addresses are determined.
+{{< /idea >}}
+
+#### Example using X-Forwarded-For capability with httpbin
+
+1. Specify `numTrustedProxies` as 2 either through `MeshConfig` or an `proxy.istop/io/config` annotation. If using `MeshConfig` run the following command to create a file named `topology.yaml` and apply it to your cluster:
+
+    {{< text bash >}}
+    $ cat <<'EOF' > topology.yaml
+      apiVersion: install.istio.io/v1alpha1
+      kind: IstioOperator
+      spec:
+        meshConfig:
+          defaultConfig:
+            gatewayTopology:
+              numTrustedProxies: 2
+      EOF
+    $ istioctl manifest apply -f topology.yaml
```

Comment:
Add a note about picking the correct profile here.

Reply:
Add a step about restarting Gateway pod for this to take effect.

Reply:
That's only true if you have an already existing gateway pod, right? I can put that in.

Reply:
yes

Reply:
done